pandas/tests/test_strings.py

```diff
@@ -2002,6 +2002,12 @@ def test_split_to_dataframe(self):
                          5: [NA, 'not']})
         tm.assert_frame_equal(result, exp)
 
+        # make sure we propogate NaN values across all columns
+        s = Series(["foo,bar,baz", NA])
+        result = s.str.split(",", expand=True)
+        exp = DataFrame([["foo", "bar", "baz"], [NA, NA, NA]])
+        tm.assert_frame_equal(result, exp)
```

Comment:
make this a separate test. this doesn't fail currently (because of the assert_frame_equal comparisons), so add an additional check that assert all nans are actually nan